src/FastIDs.TypeId/TypeId.Core/TypeIdDecoded.cs

```diff
@@ -5,9 +5,23 @@
 namespace FastIDs.TypeId;
 
 [StructLayout(LayoutKind.Auto)]
-public readonly struct TypeIdDecoded : IEquatable<TypeIdDecoded>, ISpanFormattable, IUtf8SpanFormattable
+public readonly struct TypeIdDecoded : IEquatable<TypeIdDecoded>, ISpanFormattable, IUtf8SpanFormattable, IComparable<TypeIdDecoded>, IComparable
 {
+    private static IComparer<TypeIdDecoded> ComparerValue = Comparers.Lex;
     private static readonly UuidGenerator UuidGenerator = new();
+
+    /// <summary>
+    /// The comparer that is used by <see cref="IComparable{T}"/> methods and comparison operators.
+    /// </summary>
+    /// <exception cref="ArgumentNullException">
+    /// The value is null.
+    /// </exception>
+    public static IComparer<TypeIdDecoded> Comparer
+    {
+        get => ComparerValue;
+        set => ComparerValue = value ?? throw new ArgumentNullException(nameof(value));
+        
+    }
```

Comment:
I prefer this comparer to be in `public static class Comparers` down below and named `Default`. It can also be an auto-property without a backing field and null check:
- The library supports nullable reference types so assigning null as a comparer should show a warning.
- It's highly unlikely to make this bug by accident.